ambry-frontend/src/main/java/com.github.ambry.frontend/TtlUpdateHandler.java

```diff
@@ -0,0 +1,261 @@
+     * @param blobId the {@link BlobId} to operate on
+     * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+     */
+    SecurityPostProcessRequestCallback(RestRequest restRequest, RestResponseChannel restResponseChannel, BlobId blobId,
+        Callback<Void> callback) {
+      this.restRequest = restRequest;
+      this.restResponseChannel = restResponseChannel;
+      this.blobId = blobId;
+      this.callback = callback;
+    }
+
+    @Override
+    public void onCompletion(Void result, Exception exception) {
+      long processingStartTimeMs = SystemTime.getInstance().milliseconds();
+      metrics.updateBlobTtlSecurityPostProcessRequestTimeInMs.update(processingStartTimeMs - operationStartTimeMs);
+      try {
+        if (exception == null) {
+          String serviceId = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SERVICE_ID, true);
+          router.updateBlobTtl(blobId.getID(), serviceId, Utils.Infinite_Time,
+              new RouterCallback(restRequest, restResponseChannel, callback));
+        }
+      } catch (Exception e) {
+        exception = e;
+      } finally {
+        metrics.updateBlobTtlSecurityPostProcessRequestCallbackProcessingTimeInMs.update(
+            SystemTime.getInstance().milliseconds() - processingStartTimeMs);
+        if (exception != null) {
+          callback.onCompletion(null, exception);
+        }
+      }
+    }
+  }
+
+  /**
+   * Callback for update blob TTL in the {@link Router}
+   */
+  private class RouterCallback implements Callback<Void> {
+    private final RestRequest restRequest;
+    private final RestResponseChannel restResponseChannel;
+    private final Callback<Void> callback;
+    private final long operationStartTimeMs = SystemTime.getInstance().milliseconds();
+
+    /**
+     * @param restRequest the {@link RestRequest} received
+     * @param restResponseChannel the {@link RestResponseChannel} to return the response over
+     * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+     */
+    RouterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel, Callback<Void> callback) {
+      this.restRequest = restRequest;
+      this.restResponseChannel = restResponseChannel;
+      this.callback = callback;
+    }
+
+    @Override
+    public void onCompletion(Void result, Exception exception) {
+      long processingStartTimeMs = SystemTime.getInstance().milliseconds();
+      metrics.updateBlobTtlRouterTimeInMs.update(processingStartTimeMs - operationStartTimeMs);
+      try {
+        if (exception == null) {
+          LOGGER.debug("Updated TTL of {}",
+              RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.BLOB_ID, true));
+          restResponseChannel.setHeader(RestUtils.Headers.DATE, new GregorianCalendar().getTime());
```

Comment:
I think security service might also be setting the date header

Reply:
Unable to find the one in `AmbrySecurityService` - will chat with you